### DISCUSSION TITLE
Raise Assertion error when using `until_asserted` method

### DISCUSSION THREAD

**Discussion by ppopov212** [rockem/busypie#53] [Ideas]
I think `until_asserted` is super cool and useful, we use it a lot in one of my projects. But I think it will be better to raise the underlying `AssertionError` instead of `ConditionTimeoutError` when the timeout is reached. My reasoning is that we use busypie in our testing framework and It's much better to see the Assertion error and its message instead of `ConditionalTimeoutError`. Example:

```python
from busypie import wait


def assertion_function():
    assert 1 == 2, "Some assertion message"


def test_busypie_until_asserted():
    wait().at_most(2).until_asserted(assertion_function)


```

this will produce

`busypie.awaiter.ConditionTimeoutError: Failed to meet condition of [assertion_function] within 2 seconds`

But for me, it's better and  more readable to see the assertion error

```
AssertionError: Some assertion message
assert 1 == 2
```

My naive solution is to check if the last error produced by the function passed to `unitl_asserted` is instance of an `AssertionError` and to raise the error instead of raising `ConditionalTimeoutError`

```python
 def _validate_wait_constraint(self, condition_evaluator: ConditionEvaluator, start_time: float):
        if (time.time() - start_time) > self._condition.wait_time_in_secs:
            if isinstance(self._last_error, AssertionError): # <--- handle AssertionError
                raise self._last_error

            raise busypie.ConditionTimeoutError(
                self._describe(condition_evaluator), self._condition.wait_time_in_secs) from self._last_error
```

I see that you add the last error as the cause of the `ConditionTimeoutError`, but I have to scroll through the stack trace to find the actual cause, which is not ideal if I have to examine multiple failed tests in a test report. 

What do you think?

**Comment by rockem** [comment]
I wouldn't want to throw the actual assertion error, as the main cause is actually the timeout error. 
I do think it's a good idea to add the assertion description to the error message. 
The question is, how to get this message from AssertionError
